### ISSUE TITLE
`use`ing inherent method/const gives misleading error

### ISSUE THREAD

**Issue by clarfonthey** [rust-lang/rust#148009]
### Code

```Rust
#![feature(import_trait_associated_functions)]

pub struct Test;
 
impl Test {
    pub fn method() {}
    pub const CONST: usize = 0;
}

pub use self::Test::{method, CONST};
```

### Current output

```Shell
error[E0432]: unresolved import `self::Test`
  --> src/lib.rs:10:15
   |
10 | pub use self::Test::{method, CONST};
   |               ^^^^ `Test` is a struct, not a module

For more information about this error, try `rustc --explain E0432`.
```

### Desired output

```Shell
cannot export inherent associated items, only trait associated items

(import *is* resolved, just the wrong type)
```

### Rationale and extra context

Previously, only module items could be re-exported, so, this error made sense from that perspective. However, with #134691, this no longer makes sense and is a little misleading. With the feature enabled, it should mention that this is possible with trait associated items, but not inherent associated items.

*or*, alternatively, it should be implemented for inherent associated items too.

### Rust Version

```Shell
rustc 1.92.0-nightly (57ef8d642 2025-10-15)
```

**Comment by fmease** [comment]
(emphasis mine)

> ### Desired output
> cannot **export** inherent associated items […]

Whether the `use` is an "export" (more correctly a *reexport*) (i.e., an import with a visibility greather than `self`) is immaterial, so we shouldn't mention it. "import" is sufficient.

> or, alternatively, it should be implemented for inherent associated items too.

No, that's definitely out of scope for this feature (use feature `fn_delegation` / `reuse …;` instead). An "import" is never *type-relative* (meaning it can be resolved by the name resolver alone & doesn't need type checking / inference). In `use Trait::{…};`, the `Trait` is but a name (since `trait $ident… {}`). For impls however, in general we have a full *type* (since `impl… $ty {}` or `impl… $traitref for $ty {}`). Meaning the name after a supposed (pseudo) "`use <$ty>::$ident;`" would be type-relative which would require type checking / inference + (at least a "subset" of) trait solving as you'd need to *normalize* the `$ty` and semantically match it against ~all possible impls.
